Add Reader opt-in setting to Advanced Settings

pfefferle opened the pull request:
## Proposed changes:
* Add `activitypub_reader_ui` option to Advanced Settings to enable the Reader feature.
* When enabled, automatically enables the Following UI via `pre_option_activitypub_following_ui` filter.
* When enabled, automatically enables Create Posts (storing ap_posts) via `pre_option_activitypub_create_posts` filter.
* Gate the "Social Web" dashboard page behind the new setting.

### Other information:

- [x] Have you written new tests for your changes, if applicable?

## Testing instructions:

1. Go to **Settings → ActivityPub → Advanced**.
2. Enable the "Reader" checkbox.
3. Save changes.
4. Verify that:
   - The "Social Web" page appears in the Dashboard menu.
   - The "Following User Interface" option shows as enabled (forced by Reader).
   - Posts from followed accounts are stored as `ap_post` entries.
5. Disable the "Reader" checkbox.
6. Save changes.
7. Verify that:
   - The "Social Web" page is no longer visible.
   - The "Following User Interface" can be toggled independently again.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [x] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [ ] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

Add Reader opt-in setting to Advanced Settings.

</details>

pfefferle committed:
Add Reader opt-in setting to Advanced Settings

Add activitypub_reader_ui option to enable the Reader feature.
When enabled, it automatically enables the Following UI and
Create Posts options via pre_option filters.

pfefferle committed:
Fix Jetpack integration tests to check for specific callbacks

The tests were checking if any filter was registered on
pre_option_activitypub_following_ui, but now Options class also
registers a callback on this hook. Updated tests to check for
the specific Jetpack callback instead.

pfefferle committed:
Merge branch 'add/reader' into add/reader-opt-in

pfefferle committed:
Move activitypub_reader_ui register_setting to Options class

obenland commented on includes/wp-admin/class-advanced-settings-fields.php:
```diff
@@ -79,6 +79,15 @@ public static function register_advanced_fields() {
 			array( 'label_for' => 'activitypub_following_ui' )
 		);
 
+		\add_settings_field(
```
We should probably wrap this in a version check, just like the menu item registration.

obenland commented on includes/wp-admin/class-advanced-settings-fields.php:
```diff
@@ -204,6 +213,27 @@ public static function render_following_ui_field() {
+			</label>
+		</p>
+		<p class="description">
+			Adds a "Social Web" page to your Dashboard where you can read posts and shares from accounts you follow. Also enables the Following feature.
```
```suggestion
			Adds a "Social Web" interface where you can read posts and shares from accounts you follow. Also enables the Following feature. Look for it in Dashboard > Social Web.
```
Something like that to avoid ambiguity with the Dashboard itself and clarify it's in the Dashboard menu item?

obenland commented on includes/wp-admin/class-menu.php:
```diff
@@ -32,7 +32,7 @@ public static function admin_menu() {
 		\add_action( 'load-' . $settings_page, array( Screen_Options::class, 'add_settings_list_options' ) );
 
-		if ( site_supports_blocks() && \version_compare( \get_bloginfo( 'version' ), '6.9-alpha', '>=' ) ) {
+		if ( \get_option( 'activitypub_reader_ui', '0' ) && site_supports_blocks() && \version_compare( \get_bloginfo( 'version' ), '6.9-alpha', '>=' ) ) {
```
Optional, we can probably remove the `site_supports_blocks()` check. It works with the Classic Editor plugin enabled.

pfefferle committed:
Update includes/wp-admin/class-advanced-settings-fields.php

Co-authored-by: Konstantin Obenland <obenland@gmx.de>

pfefferle committed:
Remove block support check from menu condition

Eliminates the site_supports_blocks() check from the condition that adds the Social Web dashboard page. The menu item now only depends on the 'activitypub_reader_ui' option and WordPress version.

pfefferle committed:
Add version check for Reader settings field

The 'Reader' settings field is now only added if the WordPress version is 6.9-alpha or higher. Also, the 'Create Posts' field replaces the previous unconditional 'Reader' field registration.

pfefferle committed:
Remove 'Create Posts' field from advanced settings

The 'activitypub_create_posts' settings field registration has been removed from the advanced settings fields. This streamlines the settings page and may reflect a change in feature availability or configuration approach.
